### PR TITLE
Update for j2e-pac4j

### PR DESCRIPTION
**Pull request by leleuj**
See: https://github.com/pac4j/j2e-pac4j/pull/14 & https://github.com/pac4j/j2e-pac4j-demo/pull/15

### CODE REVIEW

File: pac4j-core/src/main/java/org/pac4j/core/authorization/HasAccessRoleAuthorizer.java [hunk 1]
```diff
@@ -1,63 +0,0 @@
...
-import org.pac4j.core.util.CommonHelper;
-
-/**
- * Checks an access regarding roles (require any role OR require all roles).
- *
- * @param <U> the user profile
- * @author Jerome Leleu
- * @since 1.8.0
- */
-public class HasAccessRoleAuthorizer<U extends CommonProfile> extends OrMultiAuthorizer<U> {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/HasAccessRoleAuthorizer.java]: Remove this useless complex authorizer: either you need a requireAnyRole authorizer or a requireAllRoles authorizer...

File: pac4j-core/src/main/java/org/pac4j/core/authorization/IsAuthenticatedAuthorizer.java [hunk 2]
```diff
@@ -16,34 +16,21 @@
...
  * @since 1.8.0
  */
-public class OrMultiAuthorizer<U extends CommonProfile> implements Authorizer<U> {
-
-    protected final Authorizer<U>[] authorizers;
-
-    public OrMultiAuthorizer(final Authorizer<U>[] authorizers) {
-        this.authorizers = authorizers;
-    }
+public class IsAuthenticatedAuthorizer<U extends UserProfile> implements Authorizer<U> {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/IsAuthenticatedAuthorizer.java]: This default authorizer is necessary.

File: pac4j-core/src/main/java/org/pac4j/core/profile/ProfileManager.java [hunk 3]
```diff
@@ -15,44 +15,71 @@
...
- * <p>The constructors of the subclasses will tell which inputs are required in the specific libraries.</p>
+ * <p>This class is a generic way to manage the current user profile, i.e. the one of the current authenticated user.</p>
+ * <p>It may be partially ree-implemented for specific needs / frameworks.</p>
  *
- * 
  * @author Jerome Leleu
  * @since 1.8.0
  */
-public abstract class ProfileManager<C extends CommonProfile> {
+public class ProfileManager<U extends UserProfile> {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/profile/ProfileManager.java]: Turn this abstract class into a real implementation, the abstraction is managed by using the web context.

**Comment by mmoayyed** [comment]
:+1:

**Pull request merged**